File: generated-code/codecraft/typescript/FileReadWrite/src/model/player-view.ts
import { Entity } from "./entity";
import { EntityProperties } from "./entity-properties";
import { EntityType } from "./entity-type";
import { Player } from "./player";
import { Stream } from "../stream";

/**
 * Information available to the player
 */
export class PlayerView {
    /**
     * Your player's ID
     */
    myId: number
    /**
     * Size of the map
     */
    mapSize: number
    /**
     * Whether fog of war is enabled
     */
    fogOfWar: boolean
    /**
     * Entity properties for each entity type
     */
    entityProperties: Map<EntityType, EntityProperties>
    /**
     * Max tick count for the game
     */
    maxTickCount: number
    /**
     * Max pathfind nodes when performing pathfinding in the game simulator
     */
    maxPathfindNodes: number
    /**
     * Current tick
     */
    currentTick: number
    /**
     * List of players
     */
    players: Array<Player>
    /**
     * List of entities
     */
    entities: Array<Entity>

    constructor(myId: number, mapSize: number, fogOfWar: boolean, entityProperties: Map<EntityType, EntityProperties>, maxTickCount: number, maxPathfindNodes: number, currentTick: number, players: Array<Player>, entities: Array<Entity>) {
        this.myId = myId;
        this.mapSize = mapSize;
        this.fogOfWar = fogOfWar;
        this.entityProperties = entityProperties;
        this.maxTickCount = maxTickCount;
        this.maxPathfindNodes = maxPathfindNodes;
        this.currentTick = currentTick;
        this.players = players;
        this.entities = entities;
    }

    /**
     * Read PlayerView from input stream
     */
    static async readFrom(stream: Stream): Promise<PlayerView> {
        let myId;
        myId = await stream.readInt();
        let mapSize;
        mapSize = await stream.readInt();
        let fogOfWar;
        fogOfWar = await stream.readBool();
        let entityProperties;
        entityProperties = new Map();
        for (let entityPropertiesCount = await stream.readInt(); entityPropertiesCount > 0; entityPropertiesCount--) {
            let entityPropertiesKey;
            let entityPropertiesValue;
            entityPropertiesKey = await EntityType.readFrom(stream);
            entityPropertiesValue = await EntityProperties.readFrom(stream);
            entityProperties.set(entityPropertiesKey, entityPropertiesValue)
        }
        let maxTickCount;
        maxTickCount = await stream.readInt();
        let maxPathfindNodes;
        maxPathfindNodes = await stream.readInt();
        let currentTick;
        currentTick = await stream.readInt();
        let players;
        players = [];
        for (let playersCount = await stream.readInt(); playersCount > 0; playersCount--) {
            let playersElement;
            playersElement = await Player.readFrom(stream);
            players.push(playersElement);
        }
        let entities;
        entities = [];
        for (let entitiesCount = await stream.readInt(); entitiesCount > 0; entitiesCount--) {
            let entitiesElement;
            entitiesElement = await Entity.readFrom(stream);
            entities.push(entitiesElement);
        }
        return new PlayerView(myId, mapSize, fogOfWar, entityProperties, maxTickCount, maxPathfindNodes, currentTick, players, entities)
    }

    /**
     * Write PlayerView to output stream
     */
    async writeTo(stream: Stream) {
        let myId = this.myId;
        await stream.writeInt(myId);
        let mapSize = this.mapSize;
        await stream.writeInt(mapSize);
        let fogOfWar = this.fogOfWar;
        await stream.writeBool(fogOfWar);
        let entityProperties = this.entityProperties;
        await stream.writeInt(entityProperties.size);
        for (let [entityPropertiesKey, entityPropertiesValue] of entityProperties) {
            await entityPropertiesKey.writeTo(stream);
            await entityPropertiesValue.writeTo(stream);
        }
        let maxTickCount = this.maxTickCount;
        await stream.writeInt(maxTickCount);
        let maxPathfindNodes = this.maxPathfindNodes;
        await stream.writeInt(maxPathfindNodes);
        let currentTick = this.currentTick;
        await stream.writeInt(currentTick);
        let players = this.players;
        await stream.writeInt(players.length);
        for (let playersElement of players) {
            await playersElement.writeTo(stream);
        }
        let entities = this.entities;
        await stream.writeInt(entities.length);
        for (let entitiesElement of entities) {
            await entitiesElement.writeTo(stream);
        }
    }
}

File: generated-code/python/codecraft/FileReadWrite/model/player_view.py
from model.entity import Entity
from model.entity_properties import EntityProperties
from model.entity_type import EntityType
from model.player import Player

class PlayerView:
    """Information available to the player"""

    def __init__(self, my_id, map_size, fog_of_war, entity_properties, max_tick_count, max_pathfind_nodes, current_tick, players, entities):
        self.my_id = my_id
        """Your player's ID"""
        self.map_size = map_size
        """Size of the map"""
        self.fog_of_war = fog_of_war
        """Whether fog of war is enabled"""
        self.entity_properties = entity_properties
        """Entity properties for each entity type"""
        self.max_tick_count = max_tick_count
        """Max tick count for the game"""
        self.max_pathfind_nodes = max_pathfind_nodes
        """Max pathfind nodes when performing pathfinding in the game simulator"""
        self.current_tick = current_tick
        """Current tick"""
        self.players = players
        """List of players"""
        self.entities = entities
        """List of entities"""

    @staticmethod
    def read_from(stream):
        """Read PlayerView from input stream
        """
        my_id = stream.read_int()
        map_size = stream.read_int()
        fog_of_war = stream.read_bool()
        entity_properties = {}
        for _ in range(stream.read_int()):
            entity_properties_key = EntityType(stream.read_int())
            entity_properties_value = EntityProperties.read_from(stream)
            entity_properties[entity_properties_key] = entity_properties_value
        max_tick_count = stream.read_int()
        max_pathfind_nodes = stream.read_int()
        current_tick = stream.read_int()
        players = []
        for _ in range(stream.read_int()):
            players_element = Player.read_from(stream)
            players.append(players_element)
        entities = []
        for _ in range(stream.read_int()):
            entities_element = Entity.read_from(stream)
            entities.append(entities_element)
        return PlayerView(my_id, map_size, fog_of_war, entity_properties, max_tick_count, max_pathfind_nodes, current_tick, players, entities)

    def write_to(self, stream):
        """Write PlayerView to output stream
        """
        stream.write_int(self.my_id)
        stream.write_int(self.map_size)
        stream.write_bool(self.fog_of_war)
        stream.write_int(len(self.entity_properties))
        for key, value in self.entity_properties.items():
            stream.write_int(key)
            value.write_to(stream)
        stream.write_int(self.max_tick_count)
        stream.write_int(self.max_pathfind_nodes)
        stream.write_int(self.current_tick)
        stream.write_int(len(self.players))
        for element in self.players:
            element.write_to(stream)
        stream.write_int(len(self.entities))
        for element in self.entities:
            element.write_to(stream)

    def __repr__(self):
        return "PlayerView(" + \
            repr(self.my_id) + \
            ", " + \
            repr(self.map_size) + \
            ", " + \
            repr(self.fog_of_war) + \
            ", " + \
            repr(self.entity_properties) + \
            ", " + \
            repr(self.max_tick_count) + \
            ", " + \
            repr(self.max_pathfind_nodes) + \
            ", " + \
            repr(self.current_tick) + \
            ", " + \
            repr(self.players) + \
            ", " + \
            repr(self.entities) + \
            ")"